src/main/java/net/sf/mcf2pdf/pagebuild/PageText.java

```diff
@@ -318,7 +332,43 @@ private void setBodyStyle(String css) {
 		}
 
 	}
+	
+	// <table> margins
+	private int marginTop = 0;
+	private int marginBottom = 0;
+	private int marginLeft = 0;
+	private int marginRight = 0;
+	
+	private void setTableMargins(String css) {
+		// parse attributes out of css
+		String[] avPairs = css.split(";");
 
+		for (String avp : avPairs) {
+			avp = avp.trim();
+			if (!avp.contains(":"))
+				continue;
+			String[] av = avp.split(":");
+			if (av.length != 2)
+				continue;
+			String a = av[0].trim();
+			String v = av[1].trim();
+
+			try {
+				if ("margin-top".equalsIgnoreCase(a) && v.matches("[0-9]+px"))
+					marginTop = Integer.valueOf(v.substring(0, v.indexOf("px"))).intValue();
+				if ("margin-bottom".equalsIgnoreCase(a) && v.matches("[0-9]+px"))
+					marginBottom = Integer.valueOf(v.substring(0, v.indexOf("px"))).intValue();
+				if ("margin-left".equalsIgnoreCase(a) && v.matches("[0-9]+px"))
+					marginLeft = Integer.valueOf(v.substring(0, v.indexOf("px"))).intValue();
+				if ("margin-right".equalsIgnoreCase(a) && v.matches("[0-9]+px"))
+					marginRight = Integer.valueOf(v.substring(0, v.indexOf("px"))).intValue();
```

Comment:
You could use a tool function to avoid this repetition of similar blocks. Best would be to use Regexp Pattern matching to extract numbers.